velox/functions/prestosql/types/TimestampWithTimeZoneType.cpp

```diff
@@ -60,32 +57,31 @@ void castToTimestampWithTimeZone(
     const BaseVector& input,
     exec::EvalCtx& context,
     const SelectivityVector& rows,
-    RowVector& rowResult) {
+    VectorPtr& result) {
   VELOX_CHECK_EQ(kind, TypeKind::TIMESTAMP)
   const auto inputVector = input.as<SimpleVector<Timestamp>>();
-  castFromTimestamp(*inputVector, context, rows, rowResult);
+  castFromTimestamp(*inputVector, context, rows, result);
 }
 
 void castToTimestamp(
-    const RowVector& inputVector,
+    const BaseVector& input,
     exec::EvalCtx& context,
     const SelectivityVector& rows,
     FlatVector<Timestamp>& flatResult) {
   const auto& config = context.execCtx()->queryCtx()->queryConfig();
   const auto adjustTimestampToTimezone = config.adjustTimestampToTimezone();
-  const auto timestampVector =
-      inputVector.childAt(0)->as<SimpleVector<int64_t>>();
-  const auto timezoneVector =
-      inputVector.childAt(1)->as<SimpleVector<int16_t>>();
 
   context.applyToSelectedNoThrow(rows, [&](auto row) {
-    if (inputVector.isNullAt(row)) {
+    if (input.isNullAt(row)) {
       flatResult.setNull(row, true);
     } else {
-      Timestamp ts = Timestamp::fromMillis(timestampVector->valueAt(row));
+      auto timestampWithTimezone =
+          input.as<SimpleVector<int64_t>>()->valueAt(row);
```

Comment:
input.as<SimpleVector<int64_t>> can be made a local variable outside of the applyToSelectedNoThrow to avoid the casting per row.